packages/nx-ignore/src/index.ts

```diff
@@ -105,14 +105,18 @@ async function main() {
 
   const graphJsonPath = join(tmpdir(), '.nx-affected-graph.json');
   try {
-    execSync(
-      `npx nx affected:graph --base=${baseSha} --head=${headSha} --file=${graphJsonPath}${
-        isVerbose ? ' --verbose' : ''
-      }`,
-      {
-        cwd: root,
-      }
-    );
+    const majorVersion = parseInt(nxVersion.match(/^(\d+)\./)?.[1] ?? '0');
+    const cmd =
+      majorVersion >= 19
+        ? `npx nx graph --affected --base=${baseSha} --head=${headSha} --file=${graphJsonPath}${
+            isVerbose ? ' --verbose' : ''
+          }`
+        : `npx nx affected:graph --base=${baseSha} --head=${headSha} --file=${graphJsonPath}${
+            isVerbose ? ' --verbose' : ''
+          }`;
```

Comment:
Can we swap over to just reading from `nx show projects --affected` for this? It doesn't seem like we are using the rest of the graph and that command is made for this. Should be usable in v16+ I believe

Reply:
It was `nx show projects --affected` (actually `nx affected:apps`) a long time ago, but the problem is that any Nx or npm warnings will mess up the JSON output. The output has to be written to a file and read back to avoid any warning logs.